cloud-platform-po-linter.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/ministryofjustice/cloud-platform-environments/pkg/authenticate"
+	"github.com/prometheus-operator/prometheus-operator/pkg/admission"
+	v1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/prometheus/prometheus/model/rulefmt"
+	githubaction "github.com/sethvargo/go-githubactions"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func prDir(token, githubrepo, githubref string) ([]string, error) {
+	if token == "" {
+		return nil, errors.New("you must have a valid GitHub token")
+	}
+
+	client, err := authenticate.GitHubClient(token)
+	if err != nil {
+		return nil, err
+	}
+
+	//repo user and repo name
+	githubrepoS := strings.Split(githubrepo, "/")
+	owner := githubrepoS[0]
+	repo := githubrepoS[1]
+
+	// get pr owner
+	githubrefS := strings.Split(githubref, "/")
+	branch := githubrefS[2]
+	bid, _ := strconv.Atoi(branch)
+
+	repos, _, _ := client.PullRequests.ListFiles(context.Background(), owner, repo, bid, nil)
+
+	var fileDir []string
+	for _, repo := range repos {
+		if strings.Contains(*repo.Filename, directory) {
+			fileDir = append(fileDir, *repo.Filename)
+		}
+	}
+	return fileDir, nil
+
+}
+
+func poLint(dir []string) {
+	files := dir
+
+	for _, filename := range files {
+		log.SetPrefix(fmt.Sprintf("%s: ", filename))
+		content, err := ioutil.ReadFile(filename)
```

Comment:
ioutil was deprecated in 1.16
https://pkg.go.dev/io/ioutil

I think you can use the os package.